src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -150,6 +150,16 @@ public function getServerParameter($key, $default = '')
         return isset($this->server[$key]) ? $this->server[$key] : $default;
     }
 
+    public function switchToXMLHttpRequest()
+    {
+        $this->setServerParameter('HTTP_X-Requested-With', 'XMLHttpRequest');
```

Comment:
shouldn't it be `'HTTP_X_REQUESTED_WITH'`?

Reply:
The header name is really X-Requested-With, so no I guess.

Reply:
I think @dmaicher is right: PHP normalizes dash to low dashes when hydrating the $_SERVER parameter.

Reply:
fixed, i've tested it and @dmaicher is actually right, sorry I did just check the header name.